src/main/java/gnu/io/RXTXCommDriver.java

```diff
@@ -519,18 +531,33 @@ private void registerScannedPorts(int PortType)
 		String[] CandidateDeviceNames;
 		if (debug)
 			System.out.println("scanning device directory "+deviceDirectory+" for ports of type "+PortType);
+		
+		boolean performTestRead = true;
+		
 		if(osName.toLowerCase().indexOf("windows") != -1 )
 		{
-			String[] temp = new String[getScannedBufferSize()+3];
-			for( int i = 1; i <= getScannedBufferSize(); i++ )
-			{
-				temp[i - 1] = "COM" + i;
+			// on windows, we try to get the serial port list from the registry
+			try {
+				String[] temp = windowsGetSerialPortsFromRegistry();
+				CandidateDeviceNames=temp;
+				performTestRead = false; // disable the testRead to simply add the port to the CommPortIdentifier index
 			}
-			for( int i = 1; i <= 3; i++ )
-			{
-				temp[i + getScannedBufferSize()-1] = "LPT" + i;
+			catch (Win32Exception ex) {
+				// if something goes wrong, fallback to full scan
+				if (debug)
+					System.err.println("Error reading the registry to get port list " + ex.getMessage());
+				
+				String[] temp = new String[getScannedBufferSize()+3];
+				for( int i = 1; i <= getScannedBufferSize(); i++ )
+				{
+					temp[i - 1] = "COM" + i;
+				}
+				for( int i = 1; i <= 3; i++ )
+				{
+					temp[i + getScannedBufferSize()-1] = "LPT" + i;
```

Comment:
this FOR cycle is not copied in the upper block (lines 540-542). LPT ports should not be managed by nrjavaserial, right?